docs/rfcs/1008-nightly-build.md

```diff
@@ -0,0 +1,206 @@
+
+### Slack notification (optional)
+
+We could send a Slack notification when a nightly build finished (successfully or not)
+With the link to download the generated build.
+
+Slack provide GitHub action for that: <https://github.com/slackapi/slack-github-action>
+
+### Improve the unique dev format
+
+Currently, `releaser.py` generate a dev version by appending the `dev` part at the end of the current version before the local part.
+
+The dev part is formatted that way `dev.{number}`. The `{number}` value is calculated from the current date with the following formula:
+
+$f(year, month, day) = year * 366 + month * 31 + day$
+
+> [!NOTE]
+> If you are careful, the value $372$ ($12 * 31 = 372$) should have been used instead of $366$ (that's a mistake)
+
+For the date `2024-02-27 15:50:28+01:00` that gives `dev.740873`:
+
+$$f(2024, 2, 27) = 2024 * 366 + 2 * 31 + 27$$
+$$f(2024, 2, 27) = 740873$$
+
+The formula could be changed to:
+
+- A _timestamp_: For `2024-02-27 15:50:28+01:00` we get `dev.1709045428`
+
+- A _timestamp_ without the time info:
+
+  For `2024-02-27 15:50:28+01:00`, the value is calculated with:
+
+  $$S = 24 * 3600$$
+  $$S = 86400$$
+  $$timestamp = 1709045428$$
+  $$g(timestamp) = floor(timestamp / S)$$
+  $$g(1709045428) = 19780$$
+
+  That gives `dev.19780`, converted back to a timestamp $19780 * S = 1708992000$, that give `2024-02-27 01:00:00+01:00`
+
+- The corrected formulas using $372$:
+
+  For `2024-02-27 15:50:28+01:00`, the value is calculated with:
+
+  $$f'(year, month, day) = year * 372 + month * 31 + day$$
+  $$f'(2024, 2, 27) = 753017$$
+
+  That would give `dev.753017`
+
+- A date (`dev.YYYYMMDD`): For `2024-02-27 15:50:28+01:00` That gives `dev.20240227`
+
+## F.A.Q
+
+### Why do we need to release the nightly build? Or why we can't directly use the artifact link of a workflow?
+
+When a workflow generate artifacts, those artifacts could be directly downloaded from the workflow summary.
+
+But that has some limitations:
+
+- Those artifacts expire after a configured period (default to 90days).
+- The artifacts are wrapped in a zip file, so you need to extract them before using them.
+- You could not easily know if the workflow summary you're using is the latest available.
+- The artifacts may contain data that is not useful for you.
+
+### Why do we use the `dev` identifier for a nightly build ?
+
+We need to have a version that is compatible with [semver] and [pep440], that indicate that it's a nightly build.
+
+On [semver], we could use any verb like `nightly`, so the choice come from [pep440].
+
+Pep440 only allows the verb `dev` at the end of the version before the local identifiers ([Specify a development release in pep440])
+
+[semver]: https://semver.org/
+[pep440]: https://peps.python.org/pep-0440/
+[Specify a development release in pep440]: https://peps.python.org/pep-0440/#developmental-releases
```

Comment:
There is a note in PEP440 saying that the reference spec is in https://packaging.python.org/en/latest/specifications/version-specifiers/#developmental-releases

Reply:
They're virtually the same :shrug:

Reply:
:man_shrugging: